tests/framework/extensions/clusters/clusters.go

```diff
@@ -1,11 +1,31 @@
 package clusters
 
 import (
+	"context"
+	"fmt"
+	"strings"
+
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
+func ClusterReadyStatus(event watch.Event) (ready bool, err error) {
+	cluster := event.Object.(*apisV1.Cluster)
+
+	for _, condition := range cluster.Status.Conditions {
+		if strings.Contains(condition.Reason, "Error") {
```

Comment:
We can drop the for loop entirly we don't want to make assumptions here, this could be used in a test for recovering from an error state.